enhancements/two-node-fencing/tnf.md

```diff
@@ -0,0 +1,767 @@
+          username: <user>
+          password: <password>
+pullSecret: ''
+sshKey: ''
+```
+
+#### Installer Changes
+Aside from the changes mentioned above detailing the new install config, the installer will be also be responsible for detecting a valid two-node footprint
+(one that specifies two control-plane nodes and zero arbiters) and populating the nodes with the configuration files and resource scripts needed by pacemaker.
+t will also need update the cluster's FeatureGate CR in order to enable the `CustomNoUpgrade` feature set with the `DualReplicaTopology` feature.
+
+A cluster is assumed to be a 2-node cluster if the following statements are true:
+1. The number of control-plane replicas is set to 2
+2. The number of arbiter replicas is set to 0
+
+The number of compute nodes is also expected to be zero, but this will not be enforced.
+
+Additionally, we will enforce that 2-node clusters are only allowed on platform `None` or platform `Baremetal`. This is not a technical restriction, but
+rather one that stems from a desire to limit the number of supportable configurations. If use cases emerge, cloud support for this topology may be considered
+in the future.
+
+#### MCO Changes
+The delivery of RHEL-HA components will be opaque to the user and be delivered as an [MCO Extension](../rhcos/extensions.md) in the 4.19 timeframe.
+A switch to [MCO Layering](../ocp-coreos-layering/ocp-coreos-layering.md ) will be investigated once it is GA in a shipping version of OpenShift.
+
+Additionally, in order to ensure the cluster can upgrade safely, the MachineConfigPool `maxUnavailable` control-plane nodes will be set to 1. This should prevent
+upgrades from trying to proceed if a node is unavailable.
+
+#### Authentication Operator Changes
+The authentication operator is sensitive to the number of kube-api replicas running in the cluster for
+[test stability reasons](https://github.com/openshift/cluster-authentication-operator/commit/a08be2324f36ce89908f695a6ff3367ad86c6b78#diff-b6f4bf160fd7e801eadfbfac60b84bd00fcad21a0979c2d730549f10db015645R158-R163).
+To get around this, it runs a [readiness check](https://github.com/openshift/cluster-authentication-operator/blob/2d71f164af3f3e9c84eb40669d330df2871956f5/pkg/controllers/readiness/unsupported_override.go#L58)
+that needs to be updated to allow for a minimum of two replicas available when using the `DualReplica` infrastructure topology to ensure test stability.
+
+#### Hosted Control Plane Changes
+Two-node clusters are no compatible with hosted control planes. A check will be needed to disallow hosted control planes when the infrastructureTopology is set
+to `DualReplica`.
+
+#### OLM Filtering Changes
+Layering on top of the enhancement proposal for [Two Node with Arbiter (TNA)](../arbiter-clusters.md#olm-filter-addition), it would be ideal to include
+the `DualReplica` infrastructureTopology as an option that operators can levarage to communicate cluster-compatibility.
```

Comment:
This sounds good to me. However, I'd just like to call out that OLM itself doesn't implement or even know about the infrastructureTopology annotations. The filtering happens only in the OCP console, and there is validation (I think in CVP?) that ensures that all operators explicitly declare support (or not) for each supported infrastructure feature.

Reply:
Is the ask simply to add another filter to consolefor the `operators.openshift.io/infrastructure-features` annotation? That is a straightforward change.

Reply:
What does the infrastructure-features annotation cover?
The specifically affected field is: the `DualReplicaTopologyMode` enum of the ControlPlaneTopologyMode of the openshift infra config. https://github.com/openshift/api/pull/2196/files#diff-07b264a49084976b670fb699badaca1795027d6ea732a99226a5388104f6174fR146

Reply:
I inherited this discussion point from this [thread](https://github.com/openshift/enhancements/pull/1674/files#diff-e0198fba8283a481d7f7032cae79b58cadd8cfa685ab6ea5d8d836114ceb7182R440) in our sibling topology - Two Node OpenShift with Arbiter.

The outcome that we want is a way for operator authors to mark their operators as explicitly compatible or incompatible with one of these ControlPlaneTopologyModes.

Reply:
@jaypoulz Here is the doc around it:

https://docs.openshift.com/container-platform/4.17/operators/operator_sdk/osdk-generating-csvs.html#osdk-csv-annotations-infra_osdk-generating-csvs

They show up like this in the UI:

<img width="214" alt="Screenshot 2025-02-24 at 3 40 24 PM" src="https://github.com/user-attachments/assets/c29634da-4551-49d9-bc7d-d3f0af89e31e" />

It makes sense to me to add it... @joelanford ?

It looks like `operators.openshift.io/infrastructure-features` is actually deprecated in favor of `features.operators.openshift.io` annotations. I'll need to double check console's implementation here. But either way, the frontend work for adding a new filter for this should be straightforward. cc @jhadvig

Reply:
Another example of how they show up in the UI:
<img width="592" alt="Screenshot 2025-02-24 at 3 44 30 PM" src="https://github.com/user-attachments/assets/345ee71e-62b4-4e79-a48a-3d7037fa83b8" />

Reply:
CC @eggfoobar I think this would exactly cover our use case!